crates/extensions/tedge_flows/src/js_script.rs

```diff
@@ -116,24 +119,51 @@ impl JsScript {
         Ok(())
     }
 
+    /// Initialize the next execution time for this script's interval
+    /// Should be called after the script is loaded and interval is set
+    pub fn init_next_execution(&mut self) {
+        if !self.no_js_on_interval_fun && !self.interval.is_zero() {
+            self.next_execution = Some(Instant::now() + self.interval);
+        }
+    }
+
+    /// Check if this script should execute its interval function now
+    /// Returns true and updates next_execution if it's time to execute
+    pub fn should_execute_interval(&mut self) -> bool {
```

Comment:
It's okay for this PR to have a `should_execute_interval` function inside the `js_script` module. However, this abstraction will also be useful for rust-based flow steps as well as for [sources producing data at a regular pace](https://github.com/thin-edge/thin-edge.io/pull/3779#discussion_r2391809781).